clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareConsumerImpl.java

```diff
@@ -583,7 +583,7 @@ public synchronized ConsumerRecords<K, V> poll(final Duration timeout) {
                 final ShareFetch<K, V> fetch = pollForFetches(timer);
                 if (!fetch.isEmpty()) {
                     currentFetch = fetch;
-                    return new ConsumerRecords<>(fetch.records());
+                    return new ConsumerRecords<>(fetch.records(), Collections.emptyMap());
```

Comment:
Any reason to not leave this line as-is so that it the single-arg `ConsumerRecords` is used?

Reply:
>Any reason to not leave this line as-is so that it the single-arg ConsumerRecords is used?

since the current constructor with single-arg is marked as `@Deprecated`.

Reply:
> Trivial, but please test this in `ShareConsumerImplTest`.

Mmm. `ShareConsumerImplTest` does not test the returned records as well.